### DISCUSSION TITLE
Clearer instructions for getting file loaded onto board for auto start at boot

### DISCUSSION THREAD

**Discussion by electronicsguy** [micropython/micropython#16212] [RP2040 / Pico]
Hello,
     I just got my RP2040 tiny and was playing around with it. After fiddling around with many settings inside Thonny, I was finally able to run an example blink script on my 2040. However, for the life of me, I couldn't figure out how to actually transfer this code permanently onto the 2040. Its so much easier with ESP8266 and VScode.

I wish someone would make it clear (somewhere in the IDE, I don't know exactly) that we have to name our file as "main.py" and then "Save onto the device". 

How is a newcomer supposed to know this? I did look at some old threads: https://forum.micropython.org/viewtopic.php?t=12102
This doesn't talk about it. In another random internet website, there was some mention of a menu item saying "Device/Upload script" - But that option doesn't exist in Thonny (at least not anymore, CMIIW). 

It'd save newbie developer effort if this is highlighted somewhere in the IDE and made available. After all, that's what an IDE is for.

**Comment by GitHubsSilverBullet** [comment]
I'd say, the (by far) easiest way is to just copy main.py to the pico. 
`pyboard.py -f cp main.py :`
Done.

**Comment by andrewleech** [comment]
The official tool for uploading files is mpremote:
https://docs.micropython.org/en/latest/reference/mpremote.html

You would have to ask on thonny forums for support on that tool, it's not written by the micropython team. 

You may also like to look at Viper IDE https://viper-ide.org/
It's also third party, but written by regular micropython contributor @vshymanskyy

**Comment by fdufnews** [comment]
Well the Raspbery Pi foundation makes quite good [documentation](https://www.raspberrypi.com/documentation/microcontrollers/micropython.html) for the beginner with their products. If you search a little you find this document
[Raspberry Pi Pico-series Python SDK](https://datasheets.raspberrypi.com/pico/raspberry-pi-pico-python-sdk.pdf)
In which, at the chapter _"4.1._ Using Thonny" you read the following
![image](https://github.com/user-attachments/assets/d95ec8f0-458c-41e0-a7f5-624b55bb9711)

**Comment by Flipje1955** [comment]
On the topic of main.py and related aspects, have a look at a very informative workshop by Kevin McAleer: https://www.youtube.com/watch?v=9YvWT8bNllU.

**Comment by peterhinch** [comment]
I think this is a fair criticism. The [beginner doc](https://docs.micropython.org/en/latest/pyboard/tutorial/script.html) that explains `main.py` was written in the days when the only MP target was the Pyboard. We lack a generic "start here" doc.

**Comment by electronicsguy** [comment]
@peterhinch Thanks. I'd be happy to help in a rewrite. 

@GitHubsSilverBullet , @andrewleech , @fdufnews I'm saying for a beginner, a simple "hello world" blinky would be to take a simple example script file, run it within Thonny and have a way to "burn" it to the target (not just the pyboard). I understand that MicroPython doesn't control Thonny's interface. However, as pointed out above, an updated how-to is lacking. Asking a beginner to use CLI tools over REPL is not something I'd recommend to have a simple blinky running on a Tiny2040 independently. Yes the RaspberryPi doc mentions it so why not update it on the micropython side as well? Thanks.

**Reply by andrewleech** [reply]
As I mentioned, I'm more likely to recommend viper ide over thonny personally, but it's not my call. 
There are a number of strange things thonny does behind the scenes that cause more issues and confuses new users.

**Comment by electronicsguy** [comment]
discussed

**Comment by projectgus** [comment]
FWIW I have an open PR that improves the reference documentation about the boot process, including an explanation of boot.py and main.py that was previously only documented in the tutorial for certain boards:
https://github.com/micropython/micropython/pull/15956

This isn't an introductory tutorial though, so there might be some other "getting started" places that we should link it in or mention it.

**Comment by peterhinch** [comment]
An official beginner guide should recommend official tools (mpremote) rather than third-party IDE's. The latter present a support problem when users report problems.
